core/src/main/java/org/elasticsearch/index/query/NestedQueryParser.java

```diff
@@ -30,6 +31,10 @@
 public class NestedQueryParser implements QueryParser<NestedQueryBuilder> {
 
     private static final NestedQueryBuilder PROTOTYPE = new NestedQueryBuilder("", EmptyQueryBuilder.PROTOTYPE);
+    public static final ParseField SCORE_MODE_FIELD = new ParseField("score_mode");
+    public static final ParseField PATH_FIELD = new ParseField("path_field");
```

Comment:
I just realized with this, tying the field names we use in doXContent() to the parsers is great, but will hide accidental renames like this because at least for the generic query builder tests we the always use matching fieldnames. Maybe we need at least one test case per query that tries parsing a completly specified query from a query string using the prefered names. wdyt?